pkg/syncer/admissionhandler/cnscsi_admissionhandler.go

```diff
@@ -84,9 +84,15 @@ func (h *CSISupervisorWebhook) Handle(ctx context.Context, req admission.Request
 	log.Debugf("CNS-CSI validation webhook handler called with request: %+v", req)
 
 	resp = admission.Allowed("")
-	if featureGateTKGSHaEnabled {
-		if req.Kind.Kind == "PersistentVolumeClaim" {
-			resp = validatePVCAnnotation(ctx, req)
+	if req.Kind.Kind == "PersistentVolumeClaim" {
+		if featureGateTKGSHaEnabled {
+			resp = validatePVCAnnotationForTKGSHA(ctx, req)
+			if !resp.Allowed {
+				return
+			}
+		}
+		if featureGateVolumeHealthEnabled {
+			resp = validatePVCAnnotationForVolumeHealth(ctx, req)
 		}
 	}
 	log.Debugf("CNS-CSI validation webhook handler completed for the request: %+v", req)
```

Comment:
this debug log will never get printed when we return at line 91.

Reply:
Fixed. I have deferred the statement at the start of the method.